src/umzug.ts

```diff
@@ -414,3 +414,111 @@ export class Umzug extends EventEmitter {
 		return result;
 	}
 }
+
+import * as glob from 'glob';
+import { basename, extname } from 'path';
+
+export interface MigrationContainer {
+	up: () => Promise<unknown>;
+	down?: () => Promise<unknown>;
+}
+
+export type MigrationList = Array<{ name: string; path?: string; migration: MigrationContainer }>;
+
+export type InputMigrations<T> =
+	| {
+			/**
+			 * A glob string for migration files. Can also be in the format `[path/to/migrations/*.js', {cwd: 'some/base/dir', ignore: '**ignoreme.js' }]`
```

Comment:
```suggestion
			 * A glob string for migration files. Can also be in the format `['path/to/migrations/*.js', {cwd: 'some/base/dir', ignore: '**ignoreme.js' }]`
```

Reply:
Also, I think we should allow an array of globs as well.

Reply:
Re an array of globs, I don't think it's necessary. [glob](https://npmjs.com/package/glob) supports multiple patterns via `{folder1/migrations/*.js,folder2/migrations/*.js}`. For very complex setups, users can also use `getMigrations`:

```js
import { getMigrations, getUmzug } from 'umzug'

const m1 = getMigrations({ glob: 'folder1/*.js', ignore: '*commonIgnorePattern*' })
const m2 = getMigrations({ glob: 'folder2/*.js', ignore: ['commonIgnorePattern*', '*folder2OnlyIgnorePattern*] })

const umzug = getUmzug({ migrations: [...m1, ...m2] })
```